tests/MMS/diffusion/diffusion.cxx

```diff
@@ -77,7 +77,8 @@ int physics_init(bool UNUSED(restarting)) {
     for (int yj = mesh->ystart; yj < mesh->yend + 1; yj++){
       for (int zk = 0; zk < mesh->LocalNz; zk++) {
         output.write("Initial condition at {:d},{:d},{:d}\n", xi, yj, zk);
-        N(xi, yj, zk) = MS(0.,mesh->GlobalX(xi)*Lx,mesh->GlobalY(yj)*Ly,coord->dz*zk);
+        N(xi, yj, zk) = MS(0., mesh->GlobalX(xi) * Lx, mesh->GlobalY(yj) * Ly,
+                           coord->dz(xi, yj, zk) * zk);
```

Comment:
I assume you meant to write " if `dz` was ~not~ constant"?

Probably would be useful to ensure dz is constant, at least in z ...
However, as this is only a model, it has control over the input, and doesn't need to be generic.

Reply:
The code is correct as it is, but I think these tests should be examples of good BOUT++ style, that people might copy-paste parts of to write new tests, etc. I think now that the library allows `dz` to be non-constant, it would be good to be more consistent with the x- and y-coordinate calculations. Actually I think it would be better not to use `dz` at all (because if I remember correctly, the input expressions are in terms of normalised index-space coordinates, not real-space coordinates), maybe prefer something like `(dz / mesh->GlobalNz) * Lz`?